### PR TITLE
[Queue] Added support for receiving one message from queue

### PR DESCRIPTION
**Pull request by tasherif-msft**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/14844 and resolves https://github.com/Azure/azure-sdk-for-python/issues/14762. Many customers have reported confusion around using `receive_messages()` to pop one item from the queue.
This PR adds a new API that makes it simpler for the user to pop one element. 
Related issues:
https://github.com/Azure/azure-sdk-for-python/issues/14665
https://github.com/Azure/azure-sdk-for-python/issues/11784
https://github.com/Azure/azure-sdk-for-python/issues/14762

### CODE REVIEW

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py [hunk 1]
```diff
@@ -481,6 +481,61 @@ def send_message( # type: ignore
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace
+    def receive_one_message(self, **kwargs):
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py:485]: why not `receive_message` ?:-)

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py [hunk 2]
```diff
@@ -481,6 +481,61 @@ def send_message( # type: ignore
...
+            resolver=self.key_resolver_function)
+        try:
+            message = self._client.messages.dequeue(
+                number_of_messages=1,
+                visibilitytimeout=visibility_timeout,
+                timeout=timeout,
+                cls=self._config.message_decode_policy,
+                **kwargs
+            )
+            wrapped_message = QueueMessage._from_generated(message[0]) # pylint: disable=protected-access
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py:534]: the `dequeue` can return empty list. What would happen in that case?

**Review by kasobol-msft — COMMENTED** [review]
We should also do similar thing for `peek_messages` for completeness.

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py (hunk 1 above)
Comment (tasherif-msft) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py:485]: Haha, I felt that because we have `receive_messages()` already that `receive_one_message()` is more explicit and avoids any potential confusion. But I don't feel so strongly about it, do you feel strongly about renaming it?

**Comment by tasherif-msft** [comment]
> We should also do similar thing for `peek_messages` for completeness.

The reason I did not add a `peek_message()`/`peek_one_message()` method is because I believe the reason we wanted to create a `receieve_one_message` is due to the fact that it returns a page iterator which is not very convenient in the case of one message and creates confusion for some customers.

However, in the case of peeking the pre-exisitng `peek_messages()` is very straight forward and returns a [message(s)] depending on what the user set as `max_messages`. So if they simply pass `max_messages=1` they would get a [message1].

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py (hunk 2 above)
Comment (tasherif-msft) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py:534]: I completely forgot to consider that edge case, good catch!

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py (hunk 1 above)
Comment (kasobol-msft) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py:485]: Unless there's some pythonic reason I'd be consistent with https://github.com/Azure/azure-sdk-for-net/blob/f2b779d67932ca25a91c5c02218acaf7f1c06308/sdk/storage/Azure.Storage.Queues/src/QueueClient.cs#L1865 and https://github.com/Azure/azure-sdk-for-java/blob/6ed232113b920e957dc221ac8fcb2f05ff643b95/sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueClient.java#L454

**Comment by kasobol-msft** [comment]
> > We should also do similar thing for `peek_messages` for completeness.
> 
> The reason I did not add a `peek_message()`/`peek_one_message()` method is because I believe the reason we wanted to create a `receieve_one_message` is due to the fact that it returns a page iterator which is not very convenient in the case of one message and creates confusion for some customers.
> 
> However, in the case of peeking, the pre-exisitng `peek_messages()` is very straight forward and returns a [message(s)] depending on what the user set as `max_messages`. So if they simply pass `max_messages=1` they would get a [message1].

```
message = client.peek_message()
if message is not None:
    do_something_with(message)
else
   do_something_else()
```

How would this look like with `peek_messages` ? What's the simplest way to do this with array as a result?

**Comment by tasherif-msft** [comment]
> > > We should also do similar thing for `peek_messages` for completeness.
> > 
> > 
> > The reason I did not add a `peek_message()`/`peek_one_message()` method is because I believe the reason we wanted to create a `receieve_one_message` is due to the fact that it returns a page iterator which is not very convenient in the case of one message and creates confusion for some customers.
> > However, in the case of peeking, the pre-exisitng `peek_messages()` is very straight forward and returns a [message(s)] depending on what the user set as `max_messages`. So if they simply pass `max_messages=1` they would get a [message1].
> 
> ```
> message = client.peek_message()
> if message is not None:
>     do_something_with(message)
> else
>    do_something_else()
> ```
> 
> How would this look like with `peek_messages` ? What's the simplest way to do this with array as a result?
> In .NET I added both peek and receive after I had to write some array checking logic 10th time and that got me frustrated :)

so for your example, the current `peek_messages()` would do:
```
message = client.peek_messages(max_messages=1)
if message != []:
    do_something_with(message[0])
else:
    do_something_else()
```

**Comment by kasobol-msft** [comment]
> > > > We should also do similar thing for `peek_messages` for completeness.
> > > 
> > > 
> > > The reason I did not add a `peek_message()`/`peek_one_message()` method is because I believe the reason we wanted to create a `receieve_one_message` is due to the fact that it returns a page iterator which is not very convenient in the case of one message and creates confusion for some customers.
> > > However, in the case of peeking, the pre-exisitng `peek_messages()` is very straight forward and returns a [message(s)] depending on what the user set as `max_messages`. So if they simply pass `max_messages=1` they would get a [message1].
> > 
> > 
> > ```
> > message = client.peek_message()
> > if message is not None:
> >     do_something_with(message)
> > else
> >    do_something_else()
> > ```
> > 
> > 
> > How would this look like with `peek_messages` ? What's the simplest way to do this with array as a result?
> > In .NET I added both peek and receive after I had to write some array checking logic 10th time and that got me frustrated :)
> 
> so for your example, the current `peek_messages()` would do:
> 
> ```
> message = client.peek_messages(max_messages=1)
> if message != []:
>     do_something_with(message[0])
> else:
>     do_something_else()
> ```

that looks good. let's leave it then until we have signal from users.

File: sdk/storage/azure-storage-queue/tests/test_queue.py [hunk 3]
```diff
@@ -322,6 +322,37 @@ def test_get_messages(self, resource_group, location, storage_account, storage_a
         self.assertIsInstance(message.expires_on, datetime)
         self.assertIsInstance(message.next_visible_on, datetime)
 
+    @GlobalStorageAccountPreparer()
+    def test_receive_one_message(self, resource_group, location, storage_account, storage_account_key):
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-queue/tests/test_queue.py:326]: we should add a test case with empty queue.

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/azure-storage-queue/tests/test_queue.py (hunk 3 above)
Comment (tasherif-msft) [reply sdk/storage/azure-storage-queue/tests/test_queue.py:326]: Done!

**Review by kasobol-msft — APPROVED** [review]
